SPEC.md

```diff
@@ -0,0 +1,195 @@
+# Container Device Interface Specification
+
+- [Version](#version)
+- [Overview](#overview)
+- [General considerations](#general-considerations)
+- [CDI JSON Specification](#well-known-error-codes)
+- [CDI CLI Specification](#well-known-error-codes)
+
+## Version
+
+This is CDI **spec** version **0.1.0**.
+
+#### Released versions
+
+Released versions of the spec are available as Git tags.
+
+| Tag  | Spec Permalink   | Change |
+| -----| -----------------| -------|
+
+## Overview
+
+The _Container Device Interface_, or _CDI_ describes a mechanism for container runtimes to create containers which are able to interact with third party devices.
+
+For third party devices, it is often the case that interacting with these devices require container runtimes to expose more than a device node. For example a third party device might require you to have kernel modules loaded, host libraries mounted or specific procfs path exposed/masked.
+
+The _Container Device Interface_ describes a mechanism which allows third party vendors to perform these operations such that it doesn't require changing the container runtime.
+
+The mechanism used is a JSON file (similar the [Container Network Interface (CNI)][cni]) which allows vendors to describe the operations the container runtime should perform on the container's [OCI specification][oci].
+
+The Container Device Interface enables the following two flows:
+
+A. Device Installation
+   1. A user installs a third party device driver (and third party device) on a machine.
+   2. The device driver installation software writes a JSON file at a well known path (`/etc/cdi/vendor.json`).
+
+B. Container Runtime
+   1. A user runs a container with the argument `--device` followed by a device name.
```

Comment:
The device enablement in the container (Pod) will only happen if the argument --device is supplied. Are we removing any hook activations through ENV variables or annotations? Will there be any other triggers for hooks?

Reply:
For now the focus is on device enablement, however I'd be open to having other activation mechanisms. Maybe this is something @alban might be interested in?